docs/readmes/orc8r/dev_security.md

```diff
@@ -0,0 +1,235 @@
+
+Performed via the `RequestSign` RPC, where a gateway presents the gateway-signed challenge and a CSR, and receives back the
+Orc8r-signed CSR.
+
+- Define `signed_challenge := sign(sha256_digest( challenge ))`, computed using gateway's `gw_challenge.key`
+- Gateway sends `challenge`, `signed_challenge`, and a CSR
+- Orc8r validates request
+    - Validate `challenge` by verifying the original embedded signature and the validity window
+    - Validate `singed_challenge` by verifying the gateway's signature of `challenge`
+- Orc8r returns the CSR as the gateway's session certificates, signed by `certifier.pem`
+
+### Implementation details
+
+Some implementation-specific details, useful for debugging
+
+- Challenge is valid for short period of time (5 minutes)
+- Signed CSR includes capped validity duration (4 days)
+    - Gateways proactively reach out before expiration to re-bootstrap
+- Orc8r supports the `echo` bootstrap mode on a per-gateway basis, where, for testing purposes, gateways simply echo back the challenge
+  to receive session certs
+- Since Orc8r retains no state across the `GetChallenge` and `RequestSign` calls, gateways can make multiple `RequestSign` requests with
+  the same challenge. This is admissible because session certificates are only granted if the requesting gateway possesses the
+  requisite private key.
+
+## Debug tools
+
+This section includes a selection of notes and commands useful for debugging Orc8r from a security perspective.
+
+### Notes
+
+- [List of additional HTTP/2 debug tools](https://blog.cloudflare.com/tools-for-debugging-testing-and-using-http-2/)
+- Certificate serial number is expected to be unique, but
+  [uniqueness is *not* guaranteed](https://stackoverflow.com/questions/9104108/is-serial-number-a-unique-key-for-x509-certificate)
+
+
+### Gateway information
+
+- `show_gateway_info.py` show gateway's hardware ID and (public part of) `gw_challenge.key`
+- `checkin_cli.py` attempt to check in to Orc8r, outputting debug information on failure
+
+### CLI commands
+
+These commands have been useful for debugging Orc8r's northbound and southbound interfaces. These commands are presented targeting a local
+dev Orc8r spun up by the default docker-compose file, but with minor modifications they can target arbitrary production deployments.
+
+Utilized tools
+
+- [`curl`](https://curl.se/)
+- [`grpcurl`](https://github.com/fullstorydev/grpcurl)
+- [`s_client`](https://www.openssl.org/docs/man1.0.2/man1/openssl-s_client.html)
+
+View certificates and ACLs
+
+```bash
+# These commands are run from one of Orc8r's application containers/pods
+
+# List all registered operators, along with their ACLs and certificate serial numbers
+envdir /var/opt/magma/envdir /var/opt/magma/bin/accessc list
+
+# List all registered certificates, along with their associated identities
+envdir /var/opt/magma/envdir /var/opt/magma/bin/accessc list-certs
+```
+
+Debug northbound interface (REST API)
```

Comment:
Can you provide may be a one liner on what each of these debug command mean and what they help identify ?